### ISSUE TITLE
Product Title: Cursor becomes pointer in Editor even though it's not interactive in there

### ISSUE THREAD

**Issue by kmanijak** [woothemes/woocommerce#44277]
**Steps to reproduce:**

1. Create new post
2. Add Product Collection block and choose some collection
3. Hover over product titles

**Actual:** Cursor turns into pointer even though in Editor it's not interactive element
**Expected:** Cursor stays as default

**Tech notes:**
[Disabled component](https://developer.wordpress.org/block-editor/reference-guides/components/disabled/) component should be used for it

Reported in https://github.com/woocommerce/woocommerce/pull/44123#pullrequestreview-1854708832
